tensorflow/python/eager/context.py

```diff
@@ -1244,6 +1244,24 @@ def invoking_op_callbacks(self):
   def invoking_op_callbacks(self, value):
     self._thread_local_data.invoking_op_callbacks = value
 
+  # Parse PhysicalDevice from the device string
+  # if the device string contains subdevice type string, then parse it.
+  # if the device string doesn't contains subdevice type string, subdevice type will
+  # be the same as the device type string.
+  def _parse_physical_devices(self, devs):
+    physical_devices = []
+    for d in devs:
+        if (d.decode().count(":") == 3):
+          name = d.decode().strip(d.decode().split(":")[3]).strip(":")
+          device_type = d.decode().split(":")[1]
+          subdevice_type = d.decode().split(":")[3]
+        else:
+          name = d.decode()
+          device_type = d.decode().split(":")[1]
+          subdevice_type = d.decode().split(":")[1]
+        physical_devices.append(PhysicalDevice(name, device_type, subdevice_type))
```

Comment:
Can you also add tests for this function in `tensorflow/python/eager/context_test.py`.

Reply:
Done